includes/Story_Post_Type.php

```diff
@@ -189,6 +191,40 @@ static function( $skipped, $post ) {
 		add_filter( '_wp_post_revision_fields', [ __CLASS__, 'filter_revision_fields' ], 10, 2 );
 	}
 
+	/**
+	 * Filters the orderby query to filter first all the current user's posts and then the rest.
+	 *
+	 * @param string   $orderby Original orderby clause.
+	 * @param WP_Query $query WP_Query object.
+	 * @return string Orderby clause.
+	 */
+	public static function filter_posts_orderby( $orderby, $query ) {
+		global $wpdb;
+		if ( self::POST_TYPE_SLUG !== $query->get( 'post_type' ) ) {
+			return $orderby;
+		}
+		if ( 'author' !== $query->get( 'orderby' ) ) {
+			return $orderby;
+		}
+
+		if ( empty( $_SERVER['HTTP_REFERER'] ) ) {
+			return $orderby;
+		}
+
+		// Check the referer to ensure only `stories-dashboard` will get the filter.
+		$referer = wp_parse_url( esc_url_raw( wp_unslash( $_SERVER['HTTP_REFERER'] ) ) );
```

Comment:
`$_SERVER['HTTP_REFERER']` is probably not available in a CLI context. Maybe add an `isset()` check somewhere?

Reply:
Nevermind, I see there's an `empty` check above.